compiler/record-minmax/src/RecordMinMax.cpp

```diff
@@ -373,34 +388,106 @@ void RecordMinMax::profileData(const std::string &mode, const std::string &input
         {
           DataType dtype;
           Shape shape;
-          importer.readTensor(record_idx, input_idx, &dtype, &shape, input_data.data());
+          importer.readTensor(i, input_idx, &dtype, &shape, input_data.data());
 
           // Check the type and the shape of the input data is valid
           verifyTypeShape(input_node, dtype, shape);
         }
         else
         {
           // Skip type/shape check for raw data
-          importer.readTensor(record_idx, input_idx, input_data.data());
+          importer.readTensor(i, input_idx, input_data.data());
         }
-
-        // TODO: Input data is copied twice (file -> buffer (input_data) -> interpreter inputs)
-        //       We can redcue the copy by directly writing data from file to interpreter inputs
-        _interpreter->writeInputTensor(input_node, input_data.data(), input_data.size());
+        vector_input_data[i].emplace_back(std::move(input_data));
       }
-
-      _interpreter->interpret();
     }
 
-    std::cout << "Recording finished. Number of recorded data: " << num_records << std::endl;
+    return vector_input_data;
   }
   catch (const H5::Exception &e)
   {
     H5::Exception::printErrorStack();
     throw std::runtime_error("HDF5 error occurred.");
   }
+}
+
+void RecordMinMax::profileData(const std::string &mode, const std::string &input_data_path,
+                               float min_percentile, float max_percentile)
+{
+  const auto input_nodes = loco::input_nodes(_module->graph());
+  const auto num_inputs = input_nodes.size();
+  const auto vector_input_data = importH5Data(input_data_path);
```

Comment:
This PR loads all input data on the memory at once, which is vulnerable to OOM (out of memory).

Can you make it more memory-efficiently? E.g., processing data one by one as the previous code did.

Reply:
At least, please keep the previous behavior of the single-thread recording.

Reply:
> At least, please keep the previous behavior of the single-thread recording.

Done